### PR TITLE
fix(python/ecs-service-with-task-placement): moved placement constraint to ecs service definition block and renamed stack to comply with capacity provider naming convention

### LINKED ISSUE #628: bug(python/ecs-service-with-task-placement ): ecs-service-with-task-placement constraint.type 'distinctInstance' is not supported.
### What is the problem?

Errors occur when trying to deploy the [ecs-service-with-task-placement](https://github.com/aws-samples/aws-cdk-examples/tree/752df6302e534a7df22a809536c588deaf444c4d/python/ecs/ecs-service-with-task-placement) example.

`Resource handler returned message: "Invalid request provided: Create TaskDefinition: constraint.type 'distinctInstance' is no
t supported. (Service: AmazonECS; Status Code: 400; Error Code: ClientException`

`"Invalid request provided: CreateCapacityProvider error: The specified capacity provider name is invalid. Up to 255 characters are allowed, including letters (upper and lowercase), numbers, underscores, and hyphens. The name cannot be prefixed with "aws", "ecs", or "fargate".`

### Reproduction Steps

1. clone the aws-cdk-examples repo
2. navigate to the [python/ecs/ecs-service-with-task-placement](https://github.com/aws-samples/aws-cdk-examples/tree/752df6302e534a7df22a809536c588deaf444c4d/python/ecs/ecs-service-with-task-placement) folder
3. call `cdk deploy`

### What did you expect to happen?

Successful deployment with no errors

### What actually happened?

Two errors occur:

First error: `Resource handler returned message: "Invalid request provided: Create TaskDefinition: constraint.type 'distinctInstance' is not supported. (Service: AmazonECS; Status Code: 400; Error Code: ClientException`

Once a workaround is attempted to fix the first error, a second error is raised:
`"Invalid request provided: CreateCapacityProvider error: The specified capacity provider name is invalid. Up to 255 characters are allowed, including letters (upper and lowercase), numbers, underscores, and hyphens. The name cannot be prefixed with "aws", "ecs", or "fargate".`

### CDK CLI Version

2.13.0 (build b0b744d)

### Framework Version

_No response_

### Node.js Version

v16.14.0

### OS

Mac OS 12.2.1

### Language

Python

### Language Version

3.9.10

### Other information

The placement constraint bug is most likely due to confusing the ServicePlacementConstraint with the TaskDefinitionPlacementConstraint.

According to the docs, [TaskDefinitionPlacementConstraints](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-taskdefinitionplacementconstraint.html) only support types of `memberOf` while [ServicePlacementConstraint](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-service-placementconstraint.html) support both `memberOf` and `distinctInstances`.

In the sample code, `distinctInstance` is used in the task definition block where `memberOf` is the only placement constraint supported
https://github.com/aws-samples/aws-cdk-examples/blob/0a8a47d5f1fd20d0a53ccfa9680befa36fa4a378/python/ecs/ecs-service-with-task-placement/app.py#L37

`distinctInsance` should be used in the ECS service definition block as a placement constraint argument
https://github.com/aws-samples/aws-cdk-examples/blob/0a8a47d5f1fd20d0a53ccfa9680befa36fa4a378/python/ecs/ecs-service-with-task-placement/app.py#L54

### PR DESCRIPTION
**Pull request by its-mirus-lu**
This PR addresses #628 by renaming the stack so that the interpolated resource name for the ECS capacity provider complies with the naming requirements; this PR also moves the `distinctInstance` placement constraint to the ECS service definition block because deployment fails if it is placed in the task definition block.
 

Fixes #628 
---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: python/ecs/ecs-service-with-task-placement/app.py [hunk 1]
```diff
@@ -32,10 +32,7 @@
 
 # Create a task definition with placement constraints
 task_definition = ecs.Ec2TaskDefinition(
-    stack, "TaskDef",
-    placement_constraints=[
-        ecs.PlacementConstraint.distinct_instances()
```
Comment (its-mirus-lu) [inline python/ecs/ecs-service-with-task-placement/app.py:37]: distinct_instances doesn't belong in task definitions... only member_of is supported

File: python/ecs/ecs-service-with-task-placement/app.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 )
 
 app = App()
-stack = Stack(app, "aws-ecs-integ-ecs")
```
Comment (its-mirus-lu) [inline python/ecs/ecs-service-with-task-placement/app.py:9]: aws-ecs-integ-ecs is concatenated with the construct id in capacity provider to be used as the resource name... the resource name however can't start with 'aws'

**Review by ryparker — APPROVED** [review]
👍🏻

**Comment by its-mirus-lu** [comment]
d'oh @ryparker something funky happened and the mergify bot dismissed your previous review -- I think either closing my own comments or merging from master caused it

**Review by NGL321 — APPROVED** [review]
Approving in proxy

**Review by ryparker — APPROVED** [review]

**Pull request merged**